src/domains/wallet-app/api/payment-wallet/v1/get_wallets_by_user.xml.tpl

```diff
@@ -0,0 +1,168 @@
+<policies>
+    <inbound>
+      <set-variable  name="walletToken"  value="@(context.Request.Headers.GetValueOrDefault("Authorization", "").Replace("Bearer ",""))"  />
+      <!-- Session PM START-->
+      <send-request ignore-error="true" timeout="10" response-variable-name="pm-session-body" mode="new">
+          <set-url>@($"{{pm-host}}/pp-restapi-CD/v1/users/actions/start-session?token={(string)context.Variables["walletToken"]}")</set-url>
+          <set-method>GET</set-method>
+      </send-request>
+       <choose>
+        <when condition="@(((IResponse)context.Variables["pm-session-body"]).StatusCode != 200)">
+          <return-response>
+            <set-status code="502" reason="Bad Gateway" />
+        <set-body>
+            @{
+                JObject errorResponse = new JObject();
+                errorResponse["title"] = "Error starting session";
+                errorResponse["status"] = 502;
+                errorResponse["detail"] = "There was an error starting session for input wallet token";
+                return errorResponse.ToString();
+            }
+        </set-body>
+          </return-response>
+        </when>
+      </choose>
+      <set-variable name="pmSession" value="@(((IResponse)context.Variables["pm-session-body"]).Body.As<JObject>())" />
+    <!-- START get user wallets -->
+    <send-request ignore-error="false" timeout="10" response-variable-name="pmWalletResponse">
+        <set-url>{{pm-host}}/pp-restapi-CD/v1/wallet</set-url>
+        <set-method>GET</set-method>
+        <set-header name="Authorization" exists-action="override">
+            <value>@($"Bearer {((JObject)context.Variables["pmSession"])["data"]["sessionToken"].ToString()}")</value>
+        </set-header>
+    </send-request>
+    <choose>
+        <when condition="@(((IResponse)context.Variables["pmWalletResponse"]).StatusCode != 200)">
+            <return-response>
+            <set-status code="502" reason="Bad Gateway" />
+            <set-body>
+                @{
+                    JObject errorResponse = new JObject();
+                    errorResponse["title"] = "Error retrieving user wallet data";
+                    errorResponse["status"] = 502;
+                    errorResponse["detail"] = "There was an error retrieving user wallet data";
+                    return errorResponse.ToString();
+                }
+            </set-body>
+            </return-response>
+        </when>
+    </choose>
+    <set-variable name="pmUserWalletResponseBody" value="@(((IResponse)context.Variables["pmWalletResponse"]).Body.As<JObject>())" />
+    <set-variable name="pmUserWalletResponseBodyLength" value="@(((JArray)((JObject)context.Variables["pmUserWalletResponseBody"])["data"]).Count)" />
```

Comment:
Naming, what about

```suggestion
    <set-variable name="pmUserWalletsNumber" value="@(((JArray)((JObject)context.Variables["pmUserWalletResponseBody"])["data"]).Count)" />
```
?

Reply:
I like it! Refactored with https://github.com/pagopa/pagopa-infra/pull/1412/commits/c36a278126a2107fa31cb76e7a834299992333bb